### PR TITLE
Post-editing the attention and transformer chap

### PR DESCRIPTION
**Pull request by astonzhang**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Review by AnirudhDagar — APPROVED** [review]
@astonzhang all looks good, i've just made a couple of very small comments.

File: chapter_attention-mechanisms-and-transformers/attention-basics.md [hunk 1]
```diff
@@ -13,24 +13,24 @@ by which to *choose* which inputs to focus on.
 Typically in order to control the attention, 
 the relevant part of the model must produce 
 a *query* vector.
-This *query* vector is then typically compared to a number of 
-*key* vectors (one per input token). 
+This *query* vector is then typically compared with a number of
```
[the remark this answers is not in the document]
Comment (astonzhang) [reply chapter_attention-mechanisms-and-transformers/attention-basics.md:16]: https://preply.com/en/question/how-does-it-compare-with-or-to-50709

File: chapter_attention-mechanisms-and-transformers/index.md [hunk 2]
```diff
@@ -19,52 +18,50 @@ scaled-up implementations of classic ideas.
...
 including image recognition, object detection,
-semantic segmentation, and superresolution. 
+semantic segmentation, and superresolution :cite:`Dosovitskiy.Beyer.Kolesnikov.ea.2021,liu2021swin`. 
 Transformers are also showing up as competitive methods 
 in research papers on graph neural networks,
 electronic health records, and drug discovery.
 
-
-The core idea behind the Transformer model is the *attention mechanism*,
+The core idea behind the transformer model is the *attention mechanism*,
```
[the remark this answers is not in the document]
Comment (astonzhang) [reply chapter_attention-mechanisms-and-transformers/index.md:56]: Both are common in literature. We consistently use transformers in the book.

File: chapter_attention-mechanisms-and-transformers/large-pretraining-transformers.md [hunk 3]
```diff
@@ -132,7 +133,7 @@ Since a transformer encoder converts a sequence of input tokens
 into the same number of output representations, 
 the encoder-only mode cannot generate a sequence of arbitrary length like in machine translation. 
 As originally proposed for machine translation, 
-the transformer architecture can be outfittd with a decoder 
+the transformer architecture can be outfitted with a decoder 
 that autoregressively predicts the target sequence
```
[the remark this answers is not in the document]
Comment (astonzhang) [reply chapter_attention-mechanisms-and-transformers/large-pretraining-transformers.md:137]: Yes, it's explained in the RNN chap.

**Pull request merged**